modules/apps/transfer/types/transfer_authorization.go

```diff
@@ -0,0 +1,127 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+
+	channeltypes "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+	host "github.com/cosmos/ibc-go/v7/modules/core/24-host"
+)
+
+const gasCostPerIteration = uint64(10)
+
+var _ authz.Authorization = &TransferAuthorization{}
+
+// NewTransferAuthorization creates a new TransferAuthorization object.
+func NewTransferAuthorization(allocations ...Allocation) *TransferAuthorization {
+	return &TransferAuthorization{
+		Allocations: allocations,
+	}
+}
+
+// MsgTypeURL implements Authorization.MsgTypeURL.
+func (a TransferAuthorization) MsgTypeURL() string {
+	return sdk.MsgTypeURL(&MsgTransfer{})
+}
+
+// Accept implements Authorization.Accept.
+func (a TransferAuthorization) Accept(ctx sdk.Context, msg sdk.Msg) (authz.AcceptResponse, error) {
+	msgTransfer, ok := msg.(*MsgTransfer)
+	if !ok {
+		return authz.AcceptResponse{}, sdkerrors.Wrap(sdkerrors.ErrInvalidType, "type mismatch")
+	}
+
+	for index, allocation := range a.Allocations {
+		if allocation.SourceChannel == msgTransfer.SourceChannel && allocation.SourcePort == msgTransfer.SourcePort {
+			limitLeft, isNegative := allocation.SpendLimit.SafeSub(msgTransfer.Token)
+			if isNegative {
```

Comment:
does this support an unlimited amount?

Reply:
ideally this `SafeSub` should be skipped if the amount is the max int64. 

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/dd1265cb1dc834b129c6daf6aae7bdae28521d9e/contracts/token/ERC20/ERC20.sol#L316-L332